### ISSUE TITLE
Code Editor

### ISSUE THREAD

**Issue by LukeTOBrien** [fgilde/MudBlazor.Extensions#101]
### Feature request type

New component

### Describe the solution you'd like

I am porting a legacy app to Blazor + Mud, inside my app we have an editor (used for Markdown), the editor has been created using [CodeMirror](https://codemirror.net/).  
I just wondered that such a component would make a great little addition to your extensions project.

What do you say?

### Have you seen this feature anywhere else?

For my spercific case with Markdown, I have noticed there is a https://github.com/MyNihongo/MudBlazor.Markdown  
But this project has a dependancy on MarkDig, I don't want to compile Markdown, just an editor that will handle highlighting etc.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fgilde** [comment]
Really not sure if I want implement that. Also currently I have less time and I think personally I would prefer Monaco instead of CodeMirror. But you can easially do it as en Extensions for MudBlazor.Extensions like the https://github.com/fgilde/MudExRichTextEditor

**Comment by versile2** [comment]
That's also a fantastic example of how to create one (MudExRichTextEditor). It checks all the boxes and is not super large to get confusing.

**Comment by nefarius** [comment]
[BlazorMonaco](https://github.com/serdarciplak/BlazorMonaco) maybe?

**Comment by fgilde** [comment]
Yeah makes sense. Is also used in the demo app to showcase the json editor. (MudExStructuredDataEdit) I will check if an adaption makes sense and has benefits

**Comment by fgilde** [comment]
Not going to add a code editor component to MudBlazor.Extensions. [BlazorMonaco](https://github.com/serdarciplak/BlazorMonaco) does this well and is what the MudEx demo uses for its JSON editor, so the recommendation is to use it directly.
